apps/src/lab2/levelEditors/exemplar/ExemplarSettings.tsx

```diff
@@ -0,0 +1,231 @@
+import {BodyThreeText} from '@code-dot-org/component-library/typography';
+import React, {useState} from 'react';
+
+import {MusicExemplarSettings} from '@cdo/apps/music/types';
+import CollapsibleSection from '@cdo/apps/templates/CollapsibleSection';
+
+import {AppName, ExemplarSettings} from '../../types';
+
+import moduleStyles from './exemplar-settings.module.scss';
+
+const AppExemplarSupport: {[key in AppName]?: boolean} = {
+  music: true,
+};
+const AppPlayerSupport: {[key in AppName]?: boolean} = {
+  music: true,
+};
+
+interface ExemplarSettingsProps {
+  initialExemplarSettings: ExemplarSettings;
+  exemplarDefined: boolean;
+  appName: AppName;
+}
+
+// Default values for the validation part.
+const defaultExemplarValidationSettings: ExemplarSettings = {
+  validationSuccessMessage: '',
+  validationFailureMessage: '',
+  validationEnabled: false,
+};
+
+// Default values for the player part.
+const defaultMusicExemplarSettings: MusicExemplarSettings = {
+  ...defaultExemplarValidationSettings,
+  playerTitle: 'Example',
+  playerEnabled: false,
+};
+
+const ExemplarSettings: React.FunctionComponent<ExemplarSettingsProps> = ({
+  initialExemplarSettings,
+  exemplarDefined,
+  appName,
+}) => {
+  const appExemplarSupported = AppExemplarSupport[appName];
+  const appPlayerSupported = AppPlayerSupport[appName];
+
+  const defaultExemplarSettings = appPlayerSupported
+    ? defaultMusicExemplarSettings
+    : defaultExemplarValidationSettings;
+  const [exemplarSettings, setExemplarSettings] = useState<
+    ExemplarSettings | MusicExemplarSettings
+  >({
+    ...defaultExemplarSettings,
+    ...initialExemplarSettings,
+  });
+
+  if (!appExemplarSupported) {
+    return (
+      <div>
+        {`Exemplar settings are not available for ${appName}. Contact the engineering team for further details.`}
+      </div>
+    );
+  }
+
+  const musicExemplarSettings = appPlayerSupported
+    ? (exemplarSettings as MusicExemplarSettings)
+    : undefined;
+
+  return (
+    <div>
+      <input
+        type="hidden"
+        id="level_exemplar_settings"
+        name="level[exemplar_settings]"
+        value={JSON.stringify(exemplarSettings)}
+      />
+      <div className={moduleStyles.section}>
+        <CollapsibleSection
+          initiallyCollapsed={false}
+          headerContent="Exemplar Validation"
+        >
+          <div className={moduleStyles.row}>
+            <BodyThreeText>
+              An additional layer of validation is available for levels with
+              exemplar sources.
+              <br />
+              If checked, the playback events on the student’s timeline must
```

Comment:
This wording is pretty Music Lab specific. Might be worth just assuming this whole editor is music-lab specific in order to remove all the extra logic and casting to `MusicExemplarSettings`? Or making this more lab-agnostic and pulling out the music-specific UI (i.e. use player checkbox) into a separate component

Reply:
I created two new separate components.